### PR TITLE
feat: add `save_to_file` option to `cli state fetch`

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As part of https://github.com/ChainSafe/forest/issues/3081

Changes introduced in this pull request:

- Add an option to `state fetch` to save the DAG to a file

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: .github/workflows/forest.yml [hunk 1]
```diff
@@ -118,6 +118,14 @@ jobs:
       - name: Set permissions
         run: |
           chmod +x ~/.cargo/bin/forest*
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "^1.20"
+      - name: Install go-car
+        run: go install github.com/ipld/go-car/cmd/car@v0.0.0-20230704002448-4de031a1d8a8
```
Comment (LesnyRumcajs) [inline .github/workflows/forest.yml:125]: I like this versioning. Very agile.

Comment (hanabi1224) [reply .github/workflows/forest.yml:125]: It's recommended to use `@latest` locally but I would rather use a fixed version on CI

File: scripts/tests/calibnet_other_check.sh [hunk 2]
```diff
@@ -46,11 +46,15 @@ $FOREST_CLI_PATH sync wait # allow the node to re-sync
...
+ACTUAL_WALK=$($FOREST_CLI_PATH state fetch bafy2bzacedjq7lc42qhlk2iymcpjlanntyzdupc3ckg66gkca6plfjs5m7euo -s bafy2bzacedjq7lc42qhlk2iymcpjlanntyzdupc3ckg66gkca6plfjs5m7euo.car)
 if [[ $EXPECTED_WALK != "$ACTUAL_WALK" ]]; then
   printf "Invalid traversal:\n%s" "$ACTUAL_WALK"
   printf "Expected:\n%s" "$EXPECTED_WALK"
   exit 1
 fi
+echo "Verifying saved .car file"
+car verify bafy2bzacedjq7lc42qhlk2iymcpjlanntyzdupc3ckg66gkca6plfjs5m7euo.car
+echo "Inspecting saved .car file"
+car inspect bafy2bzacedjq7lc42qhlk2iymcpjlanntyzdupc3ckg66gkca6plfjs5m7euo.car
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply scripts/tests/calibnet_other_check.sh]: Moved out of test script

File: .github/workflows/forest.yml (hunk 1 above)
Comment (LesnyRumcajs) [reply .github/workflows/forest.yml:126]: Yeah, I was just mocking the `0.0.0` version. You cant' have any expectations in such a case. :)

**Review by LesnyRumcajs — APPROVED** [review]

File: .github/workflows/forest.yml [hunk 3]
```diff
@@ -119,8 +119,21 @@ jobs:
...
+        env:
+          CC: ""
+          CXX: ""
       - name: Other commands check
         run: ./scripts/tests/calibnet_other_check.sh
+      - name: Verify saved state root .car file
+        run: |
+          # File name is hard-coded in ./scripts/tests/calibnet_other_check.sh
+          car verify bafy2bzacedjq7lc42qhlk2iymcpjlanntyzdupc3ckg66gkca6plfjs5m7euo.car
+          car inspect bafy2bzacedjq7lc42qhlk2iymcpjlanntyzdupc3ckg66gkca6plfjs5m7euo.car
```
Comment (lemmih) [inline .github/workflows/forest.yml:136]: What's the value of `car inspect`? Can it fail even after `car verify` has succeeded? If it's only used for debugging then let's remove it.

**Review by lemmih — APPROVED** [review]

File: .github/workflows/forest.yml (hunk 3 above)
Comment (hanabi1224) [reply .github/workflows/forest.yml]: Removed

**Pull request merged**